File: apps/ios/Barns/BarnsApp/Domain/UseCases/Care/GetCareTaskDetailUseCase.swift
/// Returns a single care task by id, if present.
struct GetCareTaskDetailUseCase {
    private let repository: CareRepository

    init(repository: CareRepository) {
        self.repository = repository
    }

    func execute(id: CareTask.ID) async throws -> CareTask? {
        try await repository.careTask(id: id)
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/care/GetCareTaskDetailUseCase.kt
package com.barns.app.domain.usecase.care

import com.barns.app.domain.model.CareTask
import com.barns.app.domain.repository.CareRepository

/**
 * Returns a single care task by id, if present.
 */
class GetCareTaskDetailUseCase(
    private val repository: CareRepository,
) {
    suspend fun execute(id: String): CareTask? = repository.careTask(id)
}
